arangod/Scheduler/Scheduler.cpp

```diff
@@ -525,6 +575,30 @@ void Scheduler::shutdown() {
     std::this_thread::sleep_for(std::chrono::microseconds(20000));
   }
 
+  // no tasks are pending, bring down any threads that are sitting in run_once
+  //  a. tell io_context to release all
+  _ioContext->stop();
+
+  //  b. wait for the release to be verified
+  while (true) {
+    uint64_t const counters = _counters.load();
+
+    if (numRunning(counters) == 0 ) {
+      break;
+    }
+
+    _ioContext->post([] {
+        LOG_TOPIC(DEBUG, Logger::THREADS) << "Wakeup alarm";
+        /*wakeup call for scheduler thread*/
+      });
+
+    std::this_thread::yield();
+    // we can be quite generous here with waiting...
+    // as we are in the shutdown already, we do not care if we need to wait
+    // for a bit longer
+    std::this_thread::sleep_for(std::chrono::microseconds(20000));
```

Comment:
Why not write 20 milliseconds instead of 20000 microseconds?

Reply:
The original author of this file has consistently used microseconds throughout this file.  The lines above a copy/paste of a similar code.  I am going to leave this as microseconds to maintain consistency.